src/panel/pages/timeline/components/TimelineRow.tsx

```diff
@@ -34,6 +29,7 @@ export const TimelineRow: FC<
               position: "absolute",
               left: scale(e.timestamp),
               transform: "translateX(-50%) translateY(-50%)",
+              visibility: filter.source.includes(e.source) || "hidden",
```

Comment:
If we just don't render the row in <Timeline /> does that cause issues when toggling on/off?

Reply:
Initially I put the filtering logic in the context but then realised that filtering events from being shown on the timeline would need to be separate from filtering events that are in our state (for things such as network states).

The filter only filters the presentation of events on the timeline.